pkg/controller/nodes/task/config/config.go

```diff
@@ -45,23 +44,38 @@ type BarrierConfig struct {
 	CacheTTL  config.Duration `json:"cache-ttl" pflag:", Max duration that a barrier would be respected if the process is not restarted. This should account for time required to store the record into persistent storage (across multiple rounds."`
 }
 
+type EnabledPlugins struct {
+	DefaultPluginTasks []string `json:"default-plugins-tasks" pflag:",Tasks for which this plugin is the default implementation"`
+}
+
 type TaskPluginConfig struct {
-	EnabledPlugins []string `json:"enabled-plugins" pflag:",Plugins enabled currently"`
+	EnabledPlugins map[string]EnabledPlugins `json:"enabled-plugins" pflag:",Plugins enabled currently"`
```

Comment:
This is a "breaking" change in as far as Propeller/deployment is concerned... just a note to make sure you bump the version, and make sure the PR & the release are prefixed with "Breaking"... you can, of course, choose to leave this and add a new field and deprecate this one later.... your choice